packages/database/lib/schema.ts

```diff
@@ -175,245 +241,347 @@ export const transactionWatchers = pgTable("transaction_watchers", {
   status: pgEnum("status", TransactionWatchStatus).notNull(),
 });
 
-export const transactionWatcherTransactions = pgTable('transaction_watcher_transactions', {
-  watcherAcct: pubkey('watcher_acct').references(() => transactionWatchers.acct).notNull(),
-  txSig: transaction('tx_sig').references(() => transactions.txSig).notNull(),
-  slot: slot('slot').notNull()
-}, table => ({
-  pk: primaryKey(table.watcherAcct, table.txSig),
-  slotIdx: index('watcher_slot_index').on(table.watcherAcct, table.slot)
-}));
-
-enum IndexerImplementation {
-  AutocratV0OpenbookV2 = 'AutocratV0OpenbookV2'
+export const transactionWatcherTransactions = pgTable(
+  "transaction_watcher_transactions",
+  {
+    watcherAcct: pubkey("watcher_acct")
+      .references(() => transactionWatchers.acct)
+      .notNull(),
+    txSig: transaction("tx_sig")
+      .references(() => transactions.txSig)
+      .notNull(),
+    slot: slot("slot").notNull(),
+  },
+  (table) => ({
+    pk: primaryKey(table.watcherAcct, table.txSig),
+    slotIdx: index("watcher_slot_index").on(table.watcherAcct, table.slot),
+  })
+);
+
+export enum IndexerImplementation {
+  AutocratV0OpenbookV2 = "AutocratV0OpenbookV2",
+  AmmMarketIndexer = "AmmMarketIndexer",
+}
+export enum IndexerType {
+  TXHistory = "TXHistory",
+  AccountInfo = "AccountInfo",
 }
 
-export const indexers = pgTable('indexers', {
-  name: varchar('name', {length: 100}).primaryKey(),
-  implementation: pgEnum('implementation', IndexerImplementation).notNull(),
-  latestSlotProcessed: slot('latest_slot_processed').notNull()
+export const indexers = pgTable("indexers", {
+  name: varchar("name", { length: 100 }).primaryKey(),
+  implementation: pgEnum("implementation", IndexerImplementation).notNull(),
+  latestSlotProcessed: slot("latest_slot_processed").notNull(),
+  type: pgEnum("indexer_type", IndexerType).notNull(),
```

Comment:
So the only caveat here is that we've been naming the fields the same, dunno if that's something worth it here.

Reply:
so what would I name this in that cause? just "type"?

Reply:
indexer_type = indexerType is perfectly acceptable. I typically DON'T like using type and name as those are reserved keywords in SQL, however I've not ran into ANY issues with it so far, I'm sure I'm flirting with edge cases all the time though ;)

Reply:
I changed this to indexerType.